Make full-icu the default (#29522)
nodejs/node: BUILDING.md

>     * [Building Node.js](#building-nodejs-1)

Reviewer: there are a few places in the docs that refer to full-icu not being the default (https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_whatwg_supported_encodings). Would be good to get those fixed up in a separate commit here.

Author: @jasnell  d4c101900622dc68adc73280d589ae66aa36580b PTAL

Reviewer: Thank you!

  * [Android/Android-based devices (e.g. Firefox OS)](#androidandroid-based-devices-eg-firefox-os)
* [`Intl` (ECMA-402) support](#intl-ecma-402-support)

>   * [Trimmed: `small-icu` (English only) support](#default-small-icu-english-only-support)

Reviewer 2: ```suggestion
  * [Trimmed: `small-icu` (English only) support](#trimmed-small-icu-english-only-support)
```

>   * [Trimmed: `small-icu` (English only) support](#trimmed-small-icu-english-only-support)

Reviewer 3: It seems the TOC also needs updating now: this and the next sections neeeds to be swapped, Windows subsection for "Trimmed" needs to be added.

Reviewer 3: And maybe we also need "Unix/macOS" subsection for "Trimmed" to be consistent with the surrounding sections.

[Intl](https://github.com/nodejs/node/blob/master/doc/api/intl.md) support is
enabled by default, with English data only.

### Trimmed: `small-icu` (English only) support

> By default, only English data is included, but

Reviewer 2: This isn't the default anymore?

Reviewer 4: Maybe something like this instead?

```suggestion
 In this configuration, only English data is included, but
```

[Intl](https://github.com/nodejs/node/blob/master/doc/api/intl.md) support is

> enabled by default, with English data only.

Reviewer 3: Should this also be updated?

Reviewer 3: And also "(The embedded small-icu included in the default Node.js source does not include all locales.)" below.

Author: Good catch. I think I've addressed this now.
